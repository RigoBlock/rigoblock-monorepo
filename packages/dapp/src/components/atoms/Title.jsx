import * as React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import styles from './Title.scss'

const Title = props => (
  <h1 className={classnames(styles.title)}>{props.children}</h1>
)

Title.propTypes = {
  children: PropTypes.string.isRequired
  // classname: PropTypes.string.isRequired
}

export default Title
