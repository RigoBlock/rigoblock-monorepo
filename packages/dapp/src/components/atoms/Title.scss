.title {
  font-size: 4em;
  font-weight: lighter;
  letter-spacing: 3px;
  line-height: 1;
  margin-top: 0;
  color: var(--grey-3);

  &:after {
    content: '';
    display: block;
    width: 1em;
    border-bottom: 2px solid var(--brand);
    margin-top: -0.5em;
    margin-left: -1.1em;
  }
}
