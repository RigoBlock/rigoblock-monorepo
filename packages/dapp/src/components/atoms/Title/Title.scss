@import "../../_settings/variables.scss";

.title {
  font-weight: $font-bold;
}

.title-small {
  font-size: $medium-font-size;
}

.title-medium {
  font-size: $large-font-size;
}

.title-large {
  font-size: $xl-font-size;
}
