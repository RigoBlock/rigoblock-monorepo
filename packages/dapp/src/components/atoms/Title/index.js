export { default, TITLE_SIZES } from './Title'
