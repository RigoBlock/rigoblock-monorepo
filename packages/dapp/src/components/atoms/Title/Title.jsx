import './Title.scss'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

export const TITLE_SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large'
}

const Title = ({ size, children, className }) => {
  const classProps = classNames('title', size, className)
  return <div className={classProps}>{children}</div>
}

Title.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf([
    TITLE_SIZES.SMALL,
    TITLE_SIZES.MEDIUM,
    TITLE_SIZES.LARGE
  ])
}

Title.defaultProps = {
  size: TITLE_SIZES.MEDIUM,
  className: null
}

export default Title
