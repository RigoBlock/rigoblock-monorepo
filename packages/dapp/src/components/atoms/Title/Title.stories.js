import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { selectV2, text, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import React from 'react'
import Title, { TITLE_SIZES } from './Title'

storiesOf('Atoms/Title', module)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Title size={selectV2('Title size', TITLE_SIZES)}>
      {text('Title text', 'Example Title')}
    </Title>
  ))
