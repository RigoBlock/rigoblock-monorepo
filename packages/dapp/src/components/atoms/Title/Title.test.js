import { shallow } from 'enzyme'
import Title from './Title.jsx'
import toJson from 'enzyme-to-json'

const props = { children: 'Test title' }

describe('Title component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(Title, props)))
    ).toMatchSnapshot()
  })
})
