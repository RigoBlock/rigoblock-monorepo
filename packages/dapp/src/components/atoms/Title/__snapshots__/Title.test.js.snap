// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component renders correctly 1`] = `
<div
  className="title title-medium"
>
  Test title
</div>
`;
